internal/connection/connection_pool.go

```diff
@@ -62,11 +62,21 @@ func (cp *ConnectionPool) Delete(key string) {
 	delete(cp.pool, key)
 }
 
-// Get returns map of connections and unlock function to be called
-// after parsing the connections.
-func (cp *ConnectionPool) Get() (*map[string]*Connection, func()) {
-	rlock := cp.rwlock.RLocker()
-	rlock.Lock()
+// Get returns map of connections, filtered with given driverName and optional nodeID.
+func (cp *ConnectionPool) Get(driverName, nodeID string) map[string]*Connection {
```

Comment:
Assuming we need a drivername for all operations, there should probably be a `getByDrivername(drivername)` (internal only?) and separate `GetByNodeID(drivername, nodeID)` functions. In the future a `GetByCapability(drivername, capability)` can then easily be added without the need to modify (much) of the callers.